kodi_addon_checker/check_addon.py

```diff
@@ -384,3 +391,33 @@ def _check_dependencies(report: Report, addon_path, repo_addons):
             if LooseVersion(available_version) < LooseVersion(required_version) and (required_addon not in ignore):
                 report.add(Record(PROBLEM, "Version mismatch for addon %s. Required: %s, Available: %s "
                                   % (required_addon, required_version, available_version)))
+
+
+def get_addon_name(xml_path):
+    tree = ET.parse(xml_path).getroot()
+    return (tree.get("id"), tree.get("version"))
+
+
+def _check_for_existing_addon(report: Report, addon_path, addon_branch):
+    """Check if addon submitted already exists or not"""
+    addon_xml = os.path.join(addon_path, "addon.xml")
+    addon_name, addon_version = get_addon_name(addon_xml)
+
+    branches = ['gotham', 'helix', 'isengard', 'jarvis', 'krypton', 'leia']
+
+    for branch in branches:
+
+        branch_url = ROOT_URL.format(branch=branch)
+        repo_addons = _get_addons(branch_url)
+
+        if addon_name in repo_addons:
+            if LooseVersion(addon_version) == LooseVersion(repo_addons[addon_name]):
+                report.add(Record(PROBLEM, "%s addon already exist with current version in lower branch"
+                                  % addon_name))
+
+            else:
+                report.add(Record(INFORMATION, "%s addon also exists in %s branch but with lower version"
+                                  % (addon_name, branch)))
+                break
+        else:
+            report.add(Record(INFORMATION, "This is a new addon"))
```

Comment:
This will report multiple times and may be even report a wrong state.
E.g addon not in gotham and helix, but in isengard -> 2 times `This is a new addon` and then addon exists already

Reply:
yes thanks for pointing that out.

Reply:
yes thanks for pointing that out.

solution I can think off right now is to have a counter, initializing it with 6 i.e `branch_count=6` and instead of putting `       report.add(Record(INFORMATION, "This is a new addon"))` in else we can do `branch_count-1` and then check whether `branch_count==0` or not, outside the loop.

Reply:
simpler solution: move the report after the for loop, replace break with return